pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -229,6 +255,266 @@ func (p *Pod) Skip() bool {
 	return false
 }
 
+func (p *Pod) groupName() string {
+	return p.Object().GetLabels()[GroupNameLabel]
+}
+
+func (p *Pod) groupTotalCount() (int, error) {
+	if p.groupName() == "" {
+		return 0, fmt.Errorf("pod doesn't have a '%s' label", GroupNameLabel)
+	}
+
+	gtcAnnotation, ok := p.Object().GetAnnotations()[GroupTotalCountAnnotation]
+	if !ok {
+		return 0, fmt.Errorf("failed to extract '%s' annotation",
+			GroupTotalCountAnnotation)
+	}
+
+	gtc, err := strconv.Atoi(gtcAnnotation)
+	if err != nil {
+		return 0, err
+	}
+
+	if gtc < 1 {
+		return 0, fmt.Errorf("incorrect annotation value '%s=%s': group total count should be greater than zero",
```

Comment:
This should be part of the Pod webhook, if it isn't already.

Reply:
Yes, I'm calling this function in both validating pod webhooks:

https://github.com/kubernetes-sigs/kueue/blob/30eaeedd039fa4f520cc9735011da84bd07a7ac2/pkg/controller/jobs/pod/pod_webhook.go#L308